iron-fish/ironfish
# Rahul/ifl 2084 unspent note hashes sorted by value

## Summary

New index that keeps UNSPENT decrypted note hashes sorted by value. 

This has the effect of giving the user the best chance to include the fewest notes in a transaction. 

## Testing Plan

Unit tests + Integration tests

Manual tests: 
1. use the wallet combine notes command 
2. it should create a large note
3. when sending your next transaction that large note will be used first

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```
